sesman/chansrv/smartcard_pcsc.c

```diff
@@ -562,74 +692,95 @@ scard_function_release_context_return(void *user_data,
         return 1;
     }
     con = uds_client->con;
+    return_code = 0;
+    if (status == 0)
+    {
+        in_uint8s(in_s, 16);
+        in_uint32_le(in_s, return_code);
+    }
     out_s = trans_get_out_s(con, 8192);
-    s_push_layer(out_s, iso_hdr, 8);
-    out_uint32_le(out_s, status); /* SCARD_S_SUCCESS status */
+    init_stream(out_s, 0);
+    out_uint32_le(out_s, 0); /* hContext */
+    out_uint32_le(out_s, return_code); /* rv */
     s_mark_end(out_s);
-    bytes = (int) (out_s->end - out_s->data);
-    s_pop_layer(out_s, iso_hdr);
-    out_uint32_le(out_s, bytes - 8);
-    out_uint32_le(out_s, 0x02); /* SCARD_RELEASE_CONTEXT 0x02 */
-    return trans_force_write(con);
+    rv = trans_write_copy(con);
+    uds_client->ref_count--;
+    free_uds_client(uds_client);
+    return rv;
```

Comment:
Todo for reviewer: review this section in more detail because this diff seems odd that the steam is being reinitialized and it looks like there was data in the stream in the old version. Is this just an artifact of the diff?